sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/AutoConfiguredOpenTelemetrySdkBuilder.java

```diff
@@ -368,6 +369,12 @@ public AutoConfiguredOpenTelemetrySdkBuilder setServiceClassLoader(
     return this;
   }
 
+  public AutoConfiguredOpenTelemetrySdkBuilder setComponentLoader(ComponentLoader componentLoader) {
```

Comment:
Is the plan for the spring autoconfiguration to provide a component loader implementation which loads both from SPI and from the spring dependency injection system?

Reply:
Yes, this is the basic logic:

```java
private final SpiHelper spiHelper =
      SpiHelper.create(SpringComponentLoader.class.getClassLoader());

  public SpringComponentLoader(ApplicationContext applicationContext) {
    this.applicationContext = applicationContext;
  }

  @Override
  public <T> Iterable<T> load(Class<T> spiClass) {
    List<T> spi = spiHelper.load(spiClass);
    List<T> beans =
        applicationContext.getBeanProvider(spiClass).stream().collect(Collectors.toList());
    spi.addAll(beans);
    return spi;
  }
```

See https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/58e11116c8a5c7d8ba391283cd71452cd0ebcd10/instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/internal/SpringComponentLoader.java

Reply:
Thanks for that. Makes sense!